### DISCUSSION TITLE
Dark mode versus light mode

### DISCUSSION THREAD

**Discussion by White-Rabbit-Scientific** [beeware/toga#1664] [Ideas]
Here is a comparison on MacOS Monterey. 

Inheriting light/dark mode from the OS might be something for further down the track, however in the meantime it would be useful imo to have the ability to add "text_color" to the button widget. 

Toga in Mac light mode:
<img width="370" alt="image" src="https://user-images.githubusercontent.com/6218191/200196688-d5303ebd-6ee5-4ac1-a7aa-1b5a7c649fc1.png">

Toga in Mac dark mode:
<img width="369" alt="image" src="https://user-images.githubusercontent.com/6218191/200196708-cd5408f5-73dc-4510-9bd2-d14ebe070ceb.png">

This is what native light and dark modes looks like:

<img width="83" alt="image" src="https://user-images.githubusercontent.com/6218191/200196814-3f309c24-669b-4105-959f-650ac5e76d66.png">

<img width="84" alt="image" src="https://user-images.githubusercontent.com/6218191/200196829-d8c77fac-6369-4ca4-a02b-05ee260cf2d7.png">

**Comment by freakboy3742** [comment]
Inheriting light/dark mode is what should be happening *right now* - if it isn't, that's a bug. 

From the screenshots you've provided, there is *a* dark style being inherited; the text color is being applied correctly to the button, but it appears the background color (both of the button and the widget) isn't being set (or, is being set to the wrong value).

Although Toga supports changing text/background color on widgets, my strong advice is that it should be used *extremely* sparingly. Operating system vendors spend a lot of time developing and testing styles; and they spend a lot of time making system wide color defaults. Unless there's a *really* good, reason, users shouldn't override those defaults. "I prefer buttons to be green and have larger text" isn't a good reason, because *your* preferences won't necessarily be *my* preferences.

**Reply by White-Rabbit-Scientific** [reply]
Oh, okay, if it is inherited, then agreed, systems colors are best left to their defaults. Should this be made into a bug report, or will you handle it?

**Reply by freakboy3742** [reply]
If you can open this as a bug report with as many specifics as possible, that would be helpful.
